### PR TITLE
chore(toolchain)!: bump msrv to 1.67

### PR DESCRIPTION
**Pull request by a-kenji**
This updates the msrv to  `1.67`.

Because an optional dependency of ratatui requires it.

Alternatives: Choose to not bump the versions for optional dependencies, or pin the time version to the latest version that supports `1.65`

### CODE REVIEW

**Comment by TieWay59** [comment]
LGTM. Many CI-checks failed on this.

**Comment by orhun** [comment]
Hey, what is the exact reason for this bump? i.e. which optional dependency is that? I'm asking because I'm a proponent of avoiding MSRV bump as much as possible.

**Comment by a-kenji** [comment]
It is the current reason that the ci for `1.65` fails on some current PR's.

> error: package `time v0.3.24` cannot be built because it requires rustc 1.67.0 or newer, while the currently active rustc version is 1.65.0

I believe it is the optional `calendar` feature that pulls this dependency in.

**Review by sophacles — APPROVED** [review]

**Comment by joshka** [comment]
The N-2 policy change for the `time` crate is documented at https://github.com/time-rs/time/discussions/535.. Having read some of that thread (but not all, it's long), the arguments sound reasonable and I think we shouldn't avoid MSRV bumps.

I'd prefer to avoid the pin `time` approach (as that probably requires extra manual effort to keep abreast of any CVEs etc.), and so that the only other approach is to remove the calendar widget from the repo. I don't think that's reasonable either.

I'm approving this, but won't merge as I think it's worth hearing @orhun's perspective on this before merging.

**Review by joshka — APPROVED** [review]

**Review by orhun — APPROVED** [review]
Sounds reasonable, let's bump it up! 🆙

**Review by joshka — APPROVED** [review]

File: README.md
Comment (Valentin271) [inline README.md:153]: I think *"Since version 0.21.0"* should be updated to *0.23.0* as the MSRV will be bumped in the next release.

Comment (orhun) [reply README.md:153]: Ah, right. Feel free to submit a PR!

Comment (Valentin271) [reply README.md:153]: Done at #369 :)

**Pull request merged**
